### DISCUSSION TITLE
my_custom_setup.h with platfromio

### DISCUSSION THREAD

**Discussion by KisonHe** [Bodmer/TFT_eSPI#1145] [Q&A - General] [answered]
The readme says
```
If you load a new copy of TFT_eSPI then it will overwrite your setups if they are kept within the TFT_eSPI folder. 
One way around this is to create a new folder in your Arduino library folder called "TFT_eSPI_Setups". 
You then place your custom setup.h files in there. 
After an upgrade simply edit the User_Setup_Select.h file to point to your custom setup file e.g.:
`#include <../TFT_eSPI_Setups/my_custom_setup.h>`
```
I tried this with platformio by editing the `User_Setup_Select.h` to pick ` #include <../include/tft_eSPIConfig.h>`,
then put my `tft_eSPIConfig.h` in platformio's include folder. 
It gives me errors like 
```
In file included from .pio/libdeps/pico32/TFT_eSPI/TFT_eSPI.h:39:0,
                 from .pio/libdeps/pico32/TFT_eSPI/TFT_eSPI.cpp:17:
.pio/libdeps/pico32/TFT_eSPI/User_Setup_Select.h:23:39: fatal error: ../include/tft_eSPIConfig.h: No such file or directory
compilation terminated.
```

It seems the `User_Setup_Select.h` is inlcuded by `TFT_eSPI.h` which get included by `src/main.cpp` and `.pio/libdeps/pico32/TFT_eSPI/TFT_eSPI.cpp`
The `main.cpp` can find my `tft_eSPIConfig.h` but the `TFT_eSPI.cpp` can't.

So I changed my main.cpp to be
```
#define MAIN
#include <TFT_eSPI.h> 
```

and changed `User_Setup_Select.h`
```
#ifdef MAIN
#include <../include/tft_eSPIConfig.h>
#else
#include <../../../../include/tft_eSPIConfig.h>
#endif
```
Got it working. But this looks a little complex and pain:(
Is there any simpler and beautiful way to fix this in platformio?

**Comment by Bodmer** [comment, marked as answer]
In the Arduino environment the file path starting point for headers is the folder the header is in. So ../ moves up one directory level and expects to find your include folder, however is is not there. So you have entered the correct path by other means. As I do not use PIO I am unable to advise on this path issue.

The good news is a PIO project can have a config file dedicated to that project and then the user setup file is over-ridden. 

[See here].(https://github.com/Bodmer/TFT_eSPI/blob/master/Tools/PlatformIO/Configuring%20options.txt) .
